src/components/Documentation/Markdown/index.tsx

```diff
@@ -146,7 +146,9 @@ const Markdown: React.FC<IMarkdownProps> = ({
     touchstartXRef.current = e.changedTouches[0].screenX
   }, [])
   const onTouchEnd = useCallback(e => {
-    touchendXRef.current = e.changedTouches[0].screenX
+    const changedTouch = e.changedTouches[0]
```

Comment:
hmm .. 🤔 interesting why can it be `null`?

Reply:
I'm not sure why exactly, but the page threw when I tried clicking a details element. I would assume this check can also be done with an early return on `e.changedTouches.length === 0`.

Reply:
how did you emulate it? what OS/Browser?

Reply:
`gatsby develop` and Firefox 76.0.1 Responsive Design Mode

Reply:
Just tried Chrome and FF the master - can't reproduce it still. What is the specific action do we need for this?

Reply:
Odd, I'll try again and get more specific details on it.

Reply:
I keep getting the error on master anywhere on the mobile view with Firefox's touch simulation enabled- I don't get the issue with the touch simulation feature off.

For what it's worth, [every example of the touch APIs on MDN iterate over `changedTouches` property as a list](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events#Handling_the_end_of_a_touch), implicitly handling the case of no `changedTouches` where our code throws. I believe we're currently using the API incorrectly by failing in the case of no `changedTouches` in the `TouchList`.

Reply:
I still don't see why do we break the API tbh. Also, does it mean that the logic that depends on `touchendXRef` does not work on a mobile Firefox (I don't have it to try, unfortunately - actual one, not emulator).

Reply:
The API examples indicate that an empty `changedTouches` isn't necessarily some sort of malformed input that should be thrown on. Our current code just assumes something will always be in the `changedTouches` list and starts operating on `changedTouches[0]` even if nothing's there, which is unlike any of the examples I can find.

[Looks like empty `changedTouches` is a known firefox Responsive Display Mode issue, though](https://bugzilla.mozilla.org/show_bug.cgi?id=1615824) (Fixed in 77, I have 76- ha!), so we probably don't have to worry about Firefox Mobile being broken.

I'll update my browser and see if that fixes it. What version of Firefox are you using, @shcheklein? If it's 77+, that explains why you can't reproduce the issue.

Reply:
No, mine is 76 and I was able to reproduce it after you provided additional details. It seemed to me still that it is a FF issue, not API. The examples you brought for a reference are good but could not be considered an API spec to my mind. They make sense since in all of them you deal with multiple finger touches (not sure how would it work in our case), but they never specify explicitly that array can be empty.

So, can we close this then? :)

Reply:
Sure enough, I install my repo's Firefox Dev Edition which is on 77 and stopped having the issue! That's pretty funny that I ran into it just before fix.

I suppose we don't necessarily have to add this null check. I can't say if FF<77 RDM was the only touch handler that has this issue, but it was probably the only notably large one.